File: cmds/utility/help.ts
import { ChatInputCommandInteraction, SlashCommandBuilder } from "discord.js";
import embedBuilder from "../../lib/embedBuilder";
import { client } from "../../bot";

// create the command
const command = new SlashCommandBuilder().setName("help").setDescription("Get help with Quacks commands.");

// write the function
async function execute(interaction: ChatInputCommandInteraction) {
    const embedData = {
        title: "Quack Help",
        color: client.mainColor,
        thumbnail: interaction.client.user?.displayAvatarURL(),
        description:
            "All Quack's commands use the prefix ``/``\n\n**Commands: **[All of Quacks commands can be found here](https://quackbot.xyz/commands)\n**Need Help?** [Join our support server](https://quackbot.xyz/discord)",
        fields: [
            {
                name: "Links",
                value: "[🌐 Website](https://quackbot.xyz) | [<:invite:823987169978613851> Invite](https://quackbot.xyz/invite) | [<:discord:823989269626355793> Support](https://quackbot.xyz/discord)",
                inline: false
            }
        ]
    };

    const embed = embedBuilder(embedData as any);

    await interaction.reply({ embeds: [embed] });
}

// export the command
const data = {
    data: command,
    execute: execute
};

export { data };

File: commands/help.go
package commands

import (
	"github.com/bwmarrin/discordgo"
	"github.com/quackdiscord/bot/components"
	"github.com/quackdiscord/bot/services"
)

func init() {
	services.Commands[helpCmd.Name] = &services.Command{
		ApplicationCommand: helpCmd,
		Handler:            handleHelp,
	}
}

var helpCmd = &discordgo.ApplicationCommand{
	Type:        discordgo.ChatApplicationCommand,
	Name:        "help",
	Description: "Get some help :)",
}

func handleHelp(s *discordgo.Session, i *discordgo.InteractionCreate) *discordgo.InteractionResponse {
	embed := components.NewEmbed().
		SetTitle("Quack Help").
		SetDescription("All of Quack's commands use the `/` prefix.\n\n**Commands**: All of Quack's commands can be found [here](https://quackbot.xyz/commands)\n**Need Help?** [Join our Support Server](https://discord.gg/hUsR6fRYyE)").
		AddField("Links", "[🌐 Website](https://quackbot.xyz) | [<:invite:823987169978613851> Invite](https://quackbot.xyz/invite) | [<:discord:823989269626355793> Need Help?](https://quackbot.xyz/discord)").
		SetThumbnail("https://www.quackbot.xyz/images/png/logo.png").
		SetColor("Main").
		MessageEmbed

	return EmbedResponse(embed, false)
}
